pkg/reconciler/v1alpha1/labeler/labels.go

```diff
@@ -127,7 +127,6 @@ func (c *Reconciler) deleteLabelForOutsideOfGivenConfigurations(
 		if _, ok := configs[config.Name]; ok {
 			continue
 		}
-		delete(config.Labels, serving.RouteLabelKey)
```

Comment:
I believe this has not effect. removing it.

Reply:
Does it not have any effect because we're resetting the label below anyway? Is the commit related to the rest of the PR in any way?

Reply:
> Does it not have any effect because we're resetting the label below anyway?

in `setRouteLabelForConfiguration` below, we set labels with patch method, which has nothing to do with the content of the cached resource itself. It is not stored for further usage somewhere either. So I suppose it useless. ping @mattmoor for confirmation

> Is the commit related to the rest of the PR in any way?

Yeah, it is trying to soiled the cached configuration from shared informer here.
If this is confirmed necessary, then we'll need to make a copy before doing so.

Reply:
+1 to this analysis.  I think this is a vestige of when this was part of the Route controller, and something later in the Reconcile may have read it.

Reply:
Thanks @lichuqiang for the extended explanation. Makes perfect sense!